### PR TITLE
Allow configuration to disable endpoints in DefaultFilterChainResolverFactory

### LINKED ISSUE #712: Disabeling endpoints doesn't work in the servlet extension
DefaultFilterChainResolverFactory doesn't check if an endpoint is disabled by the user via configuration.

### PR DESCRIPTION
**Pull request by mraible**
Fixes #712.

NOTE: I'm not sure that idSite needs to be enabled/disabled in `DefaultFilterChainResolverFactory` since it seems to be configured elsewhere.

### CODE REVIEW

File: extensions/servlet/src/test/groovy/com/stormpath/sdk/servlet/config/SpecConfigVersusWebPropertiesTest.groovy [hunk 1]
```diff
@@ -73,6 +73,14 @@ class SpecConfigVersusWebPropertiesTest {
     }
 
     @Test
+    void listEnabledProperties() {
+        defaultProperties.each {
+            if (it.toString().contains('enabled')) {
+                println it
```
Comment (dogeared) [inline extensions/servlet/src/test/groovy/com/stormpath/sdk/servlet/config/SpecConfigVersusWebPropertiesTest.groovy]: This isn't actually testing anything. Maybe set the values and then make an assertion here?

File: extensions/servlet/src/test/groovy/com/stormpath/sdk/servlet/config/impl/DefaultConfigFactoryTest.groovy [hunk 2]
```diff
@@ -25,9 +25,7 @@ import org.yaml.snakeyaml.Yaml
 
 import java.lang.reflect.Field
 
-import static org.testng.Assert.assertEquals
-import static org.testng.Assert.assertNotNull
-
+import static org.testng.Assert.*
```
Comment (dogeared) [inline extensions/servlet/src/test/groovy/com/stormpath/sdk/servlet/config/impl/DefaultConfigFactoryTest.groovy]: Nit: Please revert this to the explicit imports.

File: extensions/servlet/src/test/groovy/com/stormpath/sdk/servlet/config/impl/DefaultConfigFactoryTest.groovy [hunk 3]
```diff
@@ -78,6 +76,62 @@ class DefaultConfigFactoryTest {
         assertNotNull(Classes.forName(Yaml.class.getName()))
     }
 
+    @Test
+    public void verifyAllEndpointsCanBeDisabled() {
+        /* This test verifies the following properties are read properly from stormpath.properties
```
Comment (dogeared) [inline extensions/servlet/src/test/groovy/com/stormpath/sdk/servlet/config/impl/DefaultConfigFactoryTest.groovy]: Get rid of commented out code.

**Comment by dogeared** [comment]
@mraible - a couple of nits and an update to a test. Then, I think it's GTG.

File: extensions/servlet/src/test/groovy/com/stormpath/sdk/servlet/config/impl/DefaultConfigFactoryTest.groovy (hunk 2 above)
Comment (mraible) [inline extensions/servlet/src/test/groovy/com/stormpath/sdk/servlet/config/impl/DefaultConfigFactoryTest.groovy]: I can change it, but I thought we didn't care in tests?

**Comment by mraible** [comment]
@dogeared I made updates based on your comments. Is this good to merge once CI passes?

**Comment by dogeared** [comment]
Looks like there's some merge conflicts @mraible. Resolve those and when CI passes again, please merge.

**Pull request merged**